data-types.md

```diff
@@ -5,14 +5,14 @@ toc: false
 
 CockroachDB supports the following data types. Click a type for more details.
 
-Type | Description | Example
+Type | Description | Storage Size | Example
 -----|-------------|--------
-[`INT`](int.html) | A 64-bit signed integer. | `12345`
-[`DECIMAL`](decimal.html) | An exact, fixed-point number. | `DECIMAL '1.2345'`
-[`FLOAT`](float.html) | An inexact, floating-point number. | `1.2345`
-[`BOOL`](bool.html) | A Boolean value. | `true` 
-[`DATE`](date.html) | Year, month, day. | `DATE '2016-01-25'`
-[`TIMESTAMP`](timestamp.html) | A date and time pairing. | `TIMESTAMP '2016-01-25 10:10:10'`
-[`INTERVAL`](interval.html) | A span of time. | `INTERVAL '2h30m30s'`
-[`STRING`](string.html) | A string of characters. | `'a1b2c3'`
-[`BYTES`](bytes.html) | A string of binary characters. | `b'\141\061\142\062\143\063'`
\ No newline at end of file
+[`INT`](int.html) | 64-bit signed integer. | 8 bytes | `12345`
```

Comment:
To clarify Ben's remark that "Integers (and dates) are stored in a
variable-length encoding"

There are 3 data widths involved.

1) The max width that we support, i.e. "any value that fits that width
is guaranteed to be stored by the db properly". These are the sizes that
you have in the table already. Our integers for example support values
up to 8 bytes in width. Integer, interval, date, float, timestamp, etc
all have a max size, the one in the table

Decimals, strings, byte arrays in contrast do not have a fixed max size
other than those demanded for performance.

2) the size of the data that flows back and forth between the client and
the server. This is mostly close to the values you already have in your
table.

3) the actual size of the data on disk. Like Ben said, we both do
compression on the values and add management data structures, so that
there isn't a direct correspondance between the max supported width and
the effective size on disk.

What you can highlight in the doc is that we do not support multiple
integer widths because we believe that our on-disk compression will
ensure that small integers will use a smaller size on disk, so the user
doesn't have to care about it.



-- 
Raphael 'kena' Poss